ng-dev/pr/common/validation/validation-config.ts

```diff
@@ -41,7 +43,7 @@ export abstract class PullRequestValidation {
   ) {}
 
   /** Assertion function to be defined for the specific validator. */
-  abstract assert(...parameters: unknown[]): void;
+  abstract assert(...parameters: unknown[]): Promise<void>;
```

Comment:
If you make this `void | Promise<void>`, then we don't have to make every `assert` definition async, but we can still process it as a promised result.
```suggestion
  abstract assert(...parameters: unknown[]): void | Promise<void>;
```